contrib/oauth2/mutt_oauth2.py

```diff
@@ -149,13 +149,16 @@ def writetokenfile():
     if not args.authorize:
         sys.exit('You must run script with "--authorize" at least once.')
     print('Available app and endpoint registrations:', *registrations)
-    token['registration'] = input('OAuth2 registration: ')
-    token['authflow'] = input('Preferred OAuth2 flow ("authcode" or "localhostauthcode" '
-                              'or "devicecode"): ')
+    token['registration'] = args.provider or input('OAuth2 registration: ')
+    token['authflow'] = args.authflow or input(
+        'Preferred OAuth2 flow ("authcode" or "localhostauthcode" or "devicecode"): '
+    )
     token['email'] = input('Account e-mail address: ')
     token['access_token'] = ''
     token['access_token_expiration'] = ''
     token['refresh_token'] = ''
+    token['client_id'] = args.client_id
```

Comment:
Could we make client_id and client_secret also optionally interactive? If not they are required args but the script will not warn that there is a problem if they aren't included but will create a bogus url - not a great user experience.